pandas/io/pytables.py

```diff
@@ -257,6 +257,30 @@ def _tables():
 def to_hdf(path_or_buf, key, value, mode=None, complevel=None, complib=None,
            append=None, **kwargs):
     """ store this object, close it if we opened it """
+
+    # PyTables has some limitations that we need to check for
+    if kwargs.get('format', None) == 'table':
```

Comment:
This is not the right place for this. Should go here: https://github.com/pydata/pandas/blob/master/pandas/io/pytables.py#L1546

Just add on logic to validate. This is an ``IndexCol`` at this point, so should be straightforward.